pkg/credentialprovider/config.go

```diff
@@ -64,6 +74,32 @@ func GetPreferredDockercfgPath() string {
 }
 
 func ReadDockerConfigFile() (cfg DockerConfig, err error) {
+	// Try happy path first - latest config file
```

Comment:
This function `func ReadDockerConfigFile() (cfg DockerConfig, err error)` is used to parse config which is stored in:
- `GetPreferredDockercfgPath() + "/config.json"`
- `workingDirPath + "/config.json"
- `$HOME/.docker/config.json`
- `/.docker/config.json`
- `GetPreferredDockercfgPath() + "/.dockercfg"`
- `workingDirPath + "/.dockercfg"
- `$HOME/.dockercfg`
- `/.dockercfg`

The first four one are new type of secret, and the last four one are the old type.